src/dpp/socketengines/epoll.cpp

```diff
@@ -77,14 +77,6 @@ struct DPP_EXPORT socket_engine_epoll : public socket_engine_base {
 
 	void process_events() final {
 		const int sleep_length = 1000;
-		if (sockets == 0) {
-			/* epoll_wait() on empty set waits forever (or until another thread inserts a socket into the set.
```

Comment:
note: this is an INCORRECT assumption and the cause of the bug. epoll_wait will not wait forever if called with an empty set.